share/plugins/notenames.qml

```diff
@@ -34,8 +34,8 @@ MuseScore {
    function nameChord (notes, text, small) {
       var sep = "\n";   // change to "," if you want them horizontally (anybody?)
       for (var i = 0; i < notes.length; i++) {
-         if (!notes[i].visible)
-            continue // skip invisible notes
+         if ((curScore.selection.elements.length && !notes[i].selected) || !notes[i].visible)
+            continue // skip notes that are not selected or invisible
```

Comment:
@wizofaus: please double-check.

Reply:
There are two issues as I pointed out on Discord - 1) if the first note in a chord is invisible or unselected but the 2nd isn't, it'll add a separator before the note name. 2) if there's no text to add for a particular chord (all notes invisible/unselected), it still creates an empty text element

Reply:
I don't see the 2nd issue. Maybe because an empty text is not visible?
And idea how to solve the 1st issue? Special case for 1st note?

Reply:
I don't know how you'd "see" that a empty text element is created, but programmatically it clearly is still happening (accepted, Musescore might have some special logic to ignore a request to add an empty text element, but that seems unlikely).
I suggested two possible ways to solve the leading separator issue on Discord (either only add if there's already text, or always add and strip off leading separators at the end).